rectools/models/popular.py

```diff
@@ -41,17 +43,84 @@ class Popularity(Enum):
     SUM_WEIGHT = "sum_weight"
 
 
+def _deserialize_timedelta(td: tp.Union[dict, timedelta]) -> timedelta:
+    if isinstance(td, dict):
+        return timedelta(**td)
+    return td
+
+
+def _serialize_timedelta(td: timedelta) -> dict:
+    serialized_td = {
+        key: value
+        for key, value in {"days": td.days, "seconds": td.seconds, "microseconds": td.microseconds}.items()
+        if value != 0
+    }
+    return serialized_td
+
+
+TimeDelta = tpe.Annotated[
+    timedelta,
+    PlainValidator(func=_deserialize_timedelta),
+    PlainSerializer(func=_serialize_timedelta),
+]
+
+
 class PopularModelConfig(ModelConfig):
     """Config for `PopularModel`."""
 
     popularity: Popularity = Popularity.N_USERS
-    period: tp.Optional[timedelta] = None
+    period: tp.Optional[TimeDelta] = None
     begin_from: tp.Optional[datetime] = None
     add_cold: bool = False
     inverse: bool = False
 
 
-class PopularModel(FixedColdRecoModelMixin, ModelBase):
+class PopularModelMixin:
+    """Mixin for models based on popularity."""
+
+    @classmethod
+    def _validate_popularity(
+        cls,
+        popularity: tp.Literal["n_users", "n_interactions", "mean_weight", "sum_weight"],
```

Comment:
It makes sense to separate it into a separate var like `Popularity = tp.Literal...` to avoid duplication